raft-tech/TANF-app
# 3106 reparse django action

## Summary of Changes
Pull request closes #3106 

- Added an action button to data_file admin page for reparsing fiels
- Added a script which creates a modal to confirm action
- Changed reparsing command to perform reparsing on selected number of files

## How to Test

1. Open http://localhost:3000/ and sign in.
1. submit a couple of files
2. go to /admin/data_files page
3. choose a couple of files, and use action button to start the reparse
4. you should be redirected to another page to confirm reparsing
5. push confirm to proceed, then look at the logs

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._


### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] OFA System Administrators can initiate a reparsing run from DAC
- [x] OFA System Administrators can filter/select datafiles for this action
- [x] Re-Parsing runs follow safety protocols implemented in [#3065](https://github.com/raft-tech/TANF-app/issues/3065) whether from cli cmd or DAC
- [x] Ensure Testing Checklist passes successfully
- [x]  Restrict the action such that DIGIT can not run re-parse command

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
